fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/scalar/CompoundPredicateOperator.java

```diff
@@ -25,11 +25,24 @@
 import java.util.Comparator;
 import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 import java.util.stream.Collectors;
 
 public class CompoundPredicateOperator extends PredicateOperator {
     private final CompoundType type;
 
+    // These two filed are used in NormalizePredicateRule to eliminate common CompoundPredicate
+    // For Expr tree like below, the CompoundTreeLeafNodeNumber is 5, and compoundTreeUniqueLeave's size is 4.
+    //           AND
+    //        /        \
+    //      AND         AND
+    //     /   \       /  \
+    // subT1   a+1  And  subT5
+    //               /   \
+    //             subT3  a+1
+    private int compoundTreeLeafNodeNumber;
```

Comment:
compoundTreeLeafNodeNumber is the same to compoundTreeUniqueLeaves.size(), if so, remove it and use the latter instead.

Reply:
Not really. compoundTreeLeafNodeNumber is the tree leaf nodes' size, which contains non-unique node, but compoundTreeUniqueLeaves.size() is size of unique leaf node, which is equal or greater tahn compoundTreeLeafNodeNumber. I need to compare compoundTreeLeafNodeNumber with  compoundTreeUniqueLeaves.size() to find out whether this tree can be optimized or not